src/types/heartbeat.rs

```diff
@@ -57,6 +64,19 @@ struct ConicalHeartbeat {
     pub chain_id: String,
 }
 
+impl<'a> From<&'a CanonicalHeartbeat> for chain::Id {
+    fn from(heartbeat: &CanonicalHeartbeat) -> chain::Id {
+        // TODO: non-panicking method for getting this?
+        Self::new(&heartbeat.chain_id).unwrap()
```

Comment:
As-is this is a potential DoS and we should really add non-panicking conversions, but I wanted to talk about options for these

Reply:
Actually, there is an ongoing discussion if heartbeats should be removed completely: https://github.com/tendermint/tendermint/issues/2626